packages/core/src/metro/plugin/debugIdHelper.ts

```diff
@@ -0,0 +1,279 @@
+ */
+export const DEBUG_ID_COMMENT = '//# debugId=';
+
+/**
+ * The Debug ID is injected in the virtual module as a plain string, using this prefix.
+ * It is later retrieved by searching it in the bundle, using the same prefix.
+ */
+const DEBUG_ID_METADATA_PREFIX = 'datadog-debug-id-';
+
+/**
+ * Creates a virtual module to embed a debug ID into the bundle.
+ * @param debugId - the debug ID to inject into the bundle, or a placeholder.
+ * @returns The Debug ID virtual module.
+ */
+export const createDebugIdModule = (debugId: string): DatadogDebugIdModule => {
+    let debugIdCode = createDebugIdSnippet(debugId);
+
+    return {
+        setSource: (code: string) => {
+            debugIdCode = code;
+        },
+        dependencies: new Map(),
+        getSource: () => Buffer.from(debugIdCode),
+        inverseDependencies: createCountingSet(),
+        path: DEBUG_ID_MODULE_PATH,
+        output: [
+            {
+                type: 'js/script/virtual',
+                data: {
+                    code: debugIdCode,
+                    lineCount: countLines(debugIdCode),
+                    map: []
+                }
+            }
+        ]
+    };
+};
+
+/**
+ * Injects the debug ID module into the list of pre-modules,
+ * ensuring that the prelude module (if present) stays at the top
+ * to correctly measure bundle startup time.
+ */
+export const addDebugIdModule = (
+    preModules: readonly Module<MixedOutput>[],
+    debugIdModule: DatadogDebugIdModule
+): Module<MixedOutput>[] => {
+    const result = [...preModules];
+    const hasPrelude = result.length > 0 && result[0]?.path === '__prelude__';
+
+    if (hasPrelude) {
+        result.splice(1, 0, debugIdModule);
+    } else {
+        result.unshift(debugIdModule);
+    }
+
+    return result;
+};
+
+/**
+ * Creates a minified JavaScript snippet that exposes the provided Debug ID
+ * on the global scope at runtime.
+ *
+ * @param debugId - The Debug ID to be injected into the global scope.
+ * @returns A minified JavaScript string that performs the injection.
+ */
+export const createDebugIdSnippet = (debugId: string) => {
+    return `var _datadogDebugIds,_datadogDebugIdMeta;void 0===_datadogDebugIds&&(_datadogDebugIds={});try{var stack=(new Error).stack;stack&&(_datadogDebugIds[stack]="${debugId}",_datadogDebugIdMeta="${DEBUG_ID_METADATA_PREFIX}${debugId}")}catch(e){}`;
+};
+
+/**
+ * Extracts the Debug ID from a bundle source string by locating the pattern
+ * "{@link DEBUG_ID_METADATA_PREFIX}`[DEBUG-ID]`"
+ *
+ * @param code - The source code of the bundle to search within.
+ * @returns The extracted Debug ID, or `undefined` if not found.
+ */
+export const getDebugIdFromBundleSource = (
+    code: string
+): string | undefined => {
+    const match = code.match(
+        new RegExp(
+            `${DEBUG_ID_METADATA_PREFIX}([0-9a-fA-F]{8}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{12})`
```

Comment:
Extreme nitpick:

Perhaps we could group the 3 repetitions in the middle by doing:

`[0-9a-fA-F]{8}\\b-(?:[0-9a-fA-F]{4}\\b-){3}[0-9a-fA-F]{12}`

🤓